guides/advanced_guides/filtering.md

```diff
@@ -0,0 +1,132 @@
+# Filtering
+
+A search filter is a parameter that can be applied to the search query to refine the search results of a particular field (e.g. by size, color, price, or brand).
+Multiple filters may be applied to take a broad range of documents and refine them into a more narrow selection, allowing the end user to retrieve the most relevant search results based on the criteria they've selected.
+**MeiliSearch supports filtering through a simple query language.**
+
+### The Query Language
+
+In itself the query language is very simple, and allows you to filter results on any document field. For now it only allows you to query on fields that are either `number`, `boolean` or `string`
+
+### Conditions
+
+Conditions are the primitives of query filters. They are triplets `field_id OP value` where:
+
+- `field_id` refers to the document attribute to filter on. (_e.g_ id, title...). It is either a single alphanumeric word or a quoted string: `"movie title"`, and `movie_title` are both valid `field_id`.
+- `OP` is the comparison operator, it can be one of `=`, `!=`, `<`, `>`, `>=` or `<=`.
+- `value` is the test condition for which the filter shall filter upon.
+
+The `=` and `!=` operators check for equality and difference. For strings, they are both **case insensitive**:
+
+```rust
+title = "american ninja" // matches "American Ninja"
+```
+
+::: note
+`string` values are either double-quoted, single-quoted or a single unquoted word: `title = "Scream"`, `title = 'Scream'` and `title = Scream` are all valid syntaxes, `title = The Avengers` is not.
+:::
+
+The `<`, `>`, `>=` and `<=` apply only to numerical values, and behave the standard way.
+
+::: warning
+As no specific schema is enforced at indexing, the filtering engine will try to coerce the type of `value`. This can lead to undefined behaviour when big floats are coerced into integers and reciprocally. For this reason, it is best to have homogeneous typing across fields, especially if numbers tend to become large.
+:::
+
+### Logical Connectives
+
+An arbitrary number of expressions can be connected together thanks to logical connectives. These connectives are:
+
+- `NOT`: negates the following expression, _e.g_ `NOT title = Scream`.
+- `OR`: performs a logical 'or' between two expressions, _e.g_ `title = Scream OR title = "the avengers"`
+- `AND`: performs a logical 'and' between two expressions, _e.g_ `title = Scream AND title = "the avengers"`
+
+`OR` and `AND` are left associative, meaning that the left-hand side is always evaluated first. This is particularly important for `AND`, as if the left-hand side returned `false`, the right-hand side will not be evaluated.
```

Comment:
Wouldn't it be better to move that to the "note on performance" part?

Reply:
It is more than just about performance, it explains behavior, so I think  it belongs to that section, don't you think?

Reply:
I don't think that a user care about the fact that the left or the right branch is evaluated first or second, the result will be the same, the only difference we be in terms of performances.